### PR TITLE
Fix RS1024 instances in APIView

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: src/dotnet/APIView/APIView/Languages/CodeFileBuilder.cs [hunk 1]
```diff
@@ -511,15 +511,15 @@ private CodeFileToken MapToken(ISymbol definedSymbol, SymbolDisplayPart symbolDi
...
+                (definedSymbol == null || !SymbolEqualityComparer.Default.Equals(definedSymbol, symbol)) &&
+                SymbolEqualityComparer.Default.Equals(_assembly, symbol.ContainingAssembly))
             {
                 navigateToId = symbol.GetId();
             }
 
             return new CodeFileToken()
             {
-                DefinitionId = definedSymbol?.Equals(symbol) == true ? definedSymbol.GetId() : null,
+                DefinitionId = (definedSymbol != null && SymbolEqualityComparer.Default.Equals(definedSymbol, symbol)) ? definedSymbol.GetId() : null,
```
Comment (AlexanderSher) [inline src/dotnet/APIView/APIView/Languages/CodeFileBuilder.cs:522]: You don't need null check for `definedSymbol` here, `SymbolEqualityComparer` does it:
https://github.com/dotnet/roslyn/blob/e704ca635bd6de70a0250e34c4567c7a28fa9f6d/src/Compilers/Core/Portable/Symbols/SymbolEqualityComparer.cs#L46

Comment (pakrym) [reply src/dotnet/APIView/APIView/Languages/CodeFileBuilder.cs:522]: It's important for the methods logic. `SymbolEqualityComparer.Default.Equals(null, null)` returns true and we don't want it.

Comment (AlexanderSher) [reply src/dotnet/APIView/APIView/Languages/CodeFileBuilder.cs:522]: Is it possible that `symbolDisplayPart.Symbol == null`?

Comment (pakrym) [reply src/dotnet/APIView/APIView/Languages/CodeFileBuilder.cs:522]: Yes

Comment (AlexanderSher) [reply src/dotnet/APIView/APIView/Languages/CodeFileBuilder.cs:522]: Ok

**Pull request merged**